test/pages/products/index.spec.tsx

```diff
@@ -1,21 +1,26 @@
 import { ROW_NUMBERS } from '@mocks/hooks';
 import Products from '@pages/products/index';
-import { render, screen } from '@test/utils';
+import { render, waitFor } from '@test/utils';
 
 jest.mock('@lib/hooks', () => require('@mocks/hooks'));
 
 describe('Product List', () => {
     test('renders correctly', async () => {
-        const { container } = render(<Products />);
+        const { container, getByRole } = render(<Products />);
         // Wait for table to be rendered
-        await screen.findByRole('table');
-
-        const rowsLength = screen.getAllByRole('row').length - 1; // Rows - 1 (table header)
+        await waitFor(() => getByRole('table'));
 
         expect(container.firstChild).toMatchSnapshot();
+    });
+
+    test('renders a table with correct number of rows', async () => {
+        const { getAllByRole, getByRole } = render(<Products />);
+        // Wait for table to be rendered
+        await waitFor(() => getByRole('table'));
```

Comment:
Same thing but also assign this to a const and use it in the `expect` below.
```suggestion
        const table = await screen.findByRole('table');
```

```tsx
// ...
expect(table).toBeDefined();
```